core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -77,6 +77,18 @@ class StorageStatusListener(conf: SparkConf) extends SparkListener {
       val maxMem = blockManagerAdded.maxMem
       val storageStatus = new StorageStatus(blockManagerId, maxMem)
       executorIdToStorageStatus(executorId) = storageStatus
+
+      // Try to remove the dead storage status if same executor register the block manger twice.
+      removeDeadExecutorStorageStatus(executorId)
+    }
+  }
+
+  private def removeDeadExecutorStorageStatus(executorId: String): Unit = {
+    deadExecutorStorageStatus.zipWithIndex.foreach { case (status, index) =>
```

Comment:
Just commenting on the code, not the logic: isn't this just something like `deadExecutorStorageStatus.retain(_.blockManagerId.executorId != executorId)`? I know it does something slightly different, but maybe it's desirable to clear all such matching statuses. And it's a one-liner that doesn't need a method.